### PR TITLE
chore: Remove outer directories for building CPR images.

### PR DESCRIPTION
**Pull request by abcdefgs0324**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-aiplatform/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

File: google/cloud/aiplatform/docker_utils/build.py [hunk 1]
```diff
@@ -191,23 +185,18 @@ def _prepare_entrypoint(package: Package, python_command: str = "python") -> str
 
 
 def _prepare_package_entry(package: Package) -> str:
-    """Returns the Dockerfile entries required to append at the end before entrypoint.
-
-    Including:
-    - copy the parent directory of the main executable into a docker container.
-    - inject an entrypoint that executes a script or python module inside that
-      directory.
+    """Returns the Dockerfile entries required to copy the packages to images.
```
Comment (kthytang) [inline google/cloud/aiplatform/docker_utils/build.py:188]:
```suggestion
    """Returns the Dockerfile entry required to copy the package to the image.
```

File: google/cloud/aiplatform/docker_utils/build.py [hunk 2]
```diff
@@ -191,23 +185,18 @@ def _prepare_entrypoint(package: Package, python_command: str = "python") -> str
 
 
 def _prepare_package_entry(package: Package) -> str:
```
Comment (kthytang) [inline google/cloud/aiplatform/docker_utils/build.py:187]: Does this method still need to take package as a param?

Comment (kthytang) [inline google/cloud/aiplatform/docker_utils/build.py:187]: Can we rename this to _copy_source_directory?

File: google/cloud/aiplatform/docker_utils/build.py [hunk 3]
```diff
@@ -191,23 +185,18 @@ def _prepare_entrypoint(package: Package, python_command: str = "python") -> str
...
         package (Package):
             Required. The main application copied to and run in the container.
 
     Returns:
-        The generated package related command used in Dockerfile.
+        The generated package related copy command used in Dockerfile.
     """
     copy_code = _generate_copy_command(
         ".",  # Dockefile context location has been changed to host_workdir
-        Path(package.package_path).name,
```
Comment (kthytang) [inline google/cloud/aiplatform/docker_utils/build.py:210]: What did this copy previously? This was called with our `google.cloud.aiplatform.prediction.model_server` package?

File: google/cloud/aiplatform/docker_utils/build.py [hunk 4]
```diff
@@ -117,31 +117,25 @@ def _prepare_dependency_entries(
...
+            RUN {} install --no-cache-dir {} -r {}
             """.format(
                 pip_command,
                 "--force-reinstall" if force_reinstall else "",
+                requirements_path,
             )
         )
 
     if extra_packages is not None:
         for extra in extra_packages:
```
Comment (kthytang) [inline google/cloud/aiplatform/docker_utils/build.py:131]: Nit.
```suggestion
        for package in extra_packages:
```

File: google/cloud/aiplatform/docker_utils/build.py [hunk 5]
```diff
@@ -275,14 +264,18 @@ def _get_relative_path_to_workdir(
...
     """
     if path is None:
         return None
 
+    if not Path(path).is_file():
+        raise ValueError(f'The {value_name} "{path}" must exist.')
     if not path_utils._is_relative_to(path, workdir):
         raise ValueError(f'The {value_name} "{path}" must be in "{workdir}".')
-    return Path(path).relative_to(workdir).as_posix()
+    abs_path = Path(path).expanduser().resolve()
```
Comment (kthytang) [inline google/cloud/aiplatform/docker_utils/build.py:276]: Why is this change needed?

**Review by kthytang — CHANGES_REQUESTED** [review]

File: google/cloud/aiplatform/docker_utils/build.py (hunk 3 above)
Comment (abcdefgs0324) [reply google/cloud/aiplatform/docker_utils/build.py:210]: This copied all files in the users' specified directory to a directory with the same name under the working directory of images.

File: google/cloud/aiplatform/docker_utils/build.py (hunk 2 above)
Comment (abcdefgs0324) [reply google/cloud/aiplatform/docker_utils/build.py]: Done.

Comment (abcdefgs0324) [reply google/cloud/aiplatform/docker_utils/build.py]: Removed.

File: google/cloud/aiplatform/docker_utils/build.py (hunk 1 above)
Comment (abcdefgs0324) [reply google/cloud/aiplatform/docker_utils/build.py]: Done.

File: google/cloud/aiplatform/docker_utils/build.py (hunk 4 above)
Comment (abcdefgs0324) [reply google/cloud/aiplatform/docker_utils/build.py]: Done.

File: google/cloud/aiplatform/docker_utils/build.py (hunk 5 above)
Comment (abcdefgs0324) [reply google/cloud/aiplatform/docker_utils/build.py:275]: I found there will be an issue if users use absolute paths and relative paths interchangeably. Therefore, use the absolute paths always.

**Review by kthytang — APPROVED** [review]

**Review by rosiezou — APPROVED** [review]
